### ISSUE TITLE
Enhancement: EDAM ontology tools view improvements

### ISSUE THREAD

**Issue by neoformit** [galaxyproject/galaxy#16453]
With @hechth @bgruening 
The EDAM ontology tools view is a great concept but would be much more useful if the following were addressed:

1. Text-searchable ontologies. Currently I have to scroll through a long list until I see an ontology that appeals to my analysis
1. Tree-view for ontologies. Why are we displaying hierarchically ordered data in flat lists?  
1. Better ontology annotation coverage across tool wrappers (outside the scope of this issue)
1. More comprehensive ontologies, particularly in emerging fields such as proteomics and metabolomics (also outside the scope of this issue).

I think 2) should be solved by opening an ontology tree in the main panel, as the new advanced search does. This could include some filters or a search box to find relevant ontologies quickly.

A common complaint of Galaxy users is that the service is bloated with tools. I think features like this are critical to making the service navigable and approachable for users.

![image](https://github.com/galaxyproject/galaxy/assets/42562517/27f233cb-30b9-4746-8c38-bdaae087d247)

**Comment by ahmedhamidawan** [comment]
Thanks for setting this up @neoformit and thanks for detailing the structure @mvdbeek !!

So, for the tree view, is this the tree we are referring to:
![image](https://github.com/galaxyproject/galaxy/assets/78516064/d5ef9784-159a-40f1-9fbb-e47219512da3)

which is for the `Sequence Alignment` ontology here:
![image](https://github.com/galaxyproject/galaxy/assets/78516064/b1b9939f-9445-487a-96e1-806bc03d11e2)

If so, is this the general idea we are going for (corresponding to the `Sequence Alignment` ontology for e.g., with the graph above in mind):


| Before | After|
| ------- | ------- |
| ![image](https://github.com/galaxyproject/galaxy/assets/78516064/699c8086-1d49-4875-a0a4-c0c077db3d0d) | ![image](https://github.com/galaxyproject/galaxy/assets/78516064/1ac08d4a-f957-44cd-b9a2-41e5f225550b) |

Obviously, we could try out way better ways to implement this in terms of the UI, I'm just trying to understand if this is what we mean by the tree here?

**Comment by neoformit** [comment]
Yes, that's what we meant by tree view thank you both for the details. Happy to chat if you want to brainstorm on this!

**Comment by ahmedhamidawan** [comment]
> Yes, that's what we meant by tree view thank you both for the details. Happy to chat if you want to brainstorm on this! Awesome you're already working on something, is there another issue we can link to for this?

A discussion would be great! Yes, there is an issue for refactoring the toolbox structure: https://github.com/galaxyproject/galaxy/issues/16101, although needs to be updated with what Marius described above.

**Comment by neoformit** [comment]
Ok feel free to link that info to your issue and close this one. I think your approach looks good - your drilldown UI looks like it would represent the tree structure quite concisely in the side panel.

**Comment by ahmedhamidawan** [comment]
> Ok feel free to link that info to your issue and close this one. I think your approach looks good - your drilldown UI looks like it would represent the tree structure quite concisely in the side panel.

Sounds good, maybe we can keep this one instead or keep both and then the PR will link both of them; as this one pertains more to the EDAM structure specifically, and the other one relates to the whole structure

**Comment by hechth** [comment]
@ahmedhamidawan and @mvdbeek what is the current status on this? There will be many new concepts and terms introduced to EDAM soon.

Also one thing which I saw is that currently, it is possible to choose either topics or operations. Would it be possible to have the topics as filter and then the operations? this would allow better refinement of tools and would also work better with the ontology, as some terms are more generic which are then made more specific using the related topics and the context of the tools.

**Comment by ahmedhamidawan** [comment]
Hello @hechth !

As of now, tools for given ontologies and topics are both searchable via a field in the advanced tool search (You even get a drop-down list to choose from once you have either view enabled).

Other than that, I did some work almost a year ago trying to implement sort of a nested/graph like view for EDAM in the panel, but halted work on it because _most_ tools didn't have EDAM properties.

**Comment by hechth** [comment]
So this is on hold until more tools are annotated is what you are saying? This is a bit of a chicken and egg problem.

**Comment by ahmedhamidawan** [comment]
It wasn't just that not enough tools were annotated, we were also not able to agree on an appropriate UI approach on showing the nesting tools in a tree structure in the panel. 

It's good to hear that there is more interest in this lately, and we can take your recommendations up for discussion as well (in the dev meeting as well)

**Comment by hechth** [comment]
Okay I see - it would be good to know how many of the actually used tools are annotated, because I think the fact that few tools have annotations is also related to the fact that the majority of Galaxy tools overall is not used much.

But I fully agree that there needs to be better tool annotation.
